package.json

```diff
@@ -13,7 +13,8 @@
 		"test:watch": "vitest run --coverage --watch",
 		"format": "prettier --write .",
 		"lint": "prettier --check . && eslint .",
-		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json"
+		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
+		"before-commit": "pnpm format && pnpm lint && pnpm check && pnpm test && pnpm build"
```

Comment:
`pnpm build` feels a bit excessive here. The overwhelming majority of errors I've seen on deployment are covered under the other three checks. Plus I've never found local builds to be that faithful to the actual remote deployment (Vercel, etc). Either way consider limiting this.

Reply:
This `before-commit` is a copy of our CI so you can basically "run our CI" locally before committing to succeed all checks. We can remove build, but that wouldn't guarantee that CI will pass